includes/class-newspack-ads-placements.php

```diff
@@ -311,27 +410,46 @@ public static function disable_placement( $placement_key ) {
 	}
 
 	/**
-	 * Inject Ad Unit into given placement.
+	 * Generate an ID given a list of strings as arguments.
+	 *
+	 * @param string[] $args List of strings.
+	 */
+	private static function get_id( $args ) {
+		return substr( hash( 'sha1', implode( $args ) ), 0, 10 );
+	}
+
+	/**
+	 * Inject Ad into given placement.
 	 *
 	 * @param string $placement_key Placement key.
 	 * @param string $hook_key      Optional hook key in case of multiple hooks available.
 	 */
-	public static function inject_placement_ad_unit( $placement_key, $hook_key = '' ) {
-		$placements  = self::get_placements();
-		$placement   = $placements[ $placement_key ];
-		$ad_unit_key = 'ad_unit';
+	public static function inject_placement_ad( $placement_key, $hook_key = '' ) {
+		$placements = self::get_placements();
+		$placement  = $placements[ $placement_key ];
+		$is_enabled = $placement['data']['enabled'];
+
 		if ( $hook_key ) {
-			$ad_unit_key = 'ad_unit_' . $hook_key;
+			$placement_data = $placement['data']['hooks'][ $hook_key ];
```

Comment:
Ran into a PHP error triggered by this line on both my local site and a fresh site. It seems to happen when loading up placements on the front-end which lack a `hooks` key, such as `global_above_header` which has a `hook_name` key but not a `hooks` key.

```bash
PHP Notice:  Undefined index: hooks in /Users/dkoo/Local Sites/newspack/app/public/wp-content/plugins/newspack-ads/includes/class-newspack-ads-placements.php on line 433
[08-Dec-2021 19:15:18 UTC] PHP Stack trace:
[08-Dec-2021 19:15:18 UTC] PHP   1. {main}() /Users/dkoo/Local Sites/newspack/app/public/index.php:0
[08-Dec-2021 19:15:18 UTC] PHP   2. require() /Users/dkoo/Local Sites/newspack/app/public/index.php:17
[08-Dec-2021 19:15:18 UTC] PHP   3. require_once() /Users/dkoo/Local Sites/newspack/app/public/wp-blog-header.php:19
[08-Dec-2021 19:15:18 UTC] PHP   4. include() /Users/dkoo/Local Sites/newspack/app/public/wp-includes/template-loader.php:106
[08-Dec-2021 19:15:18 UTC] PHP   5. get_header() /Users/dkoo/Local Sites/newspack/app/public/wp-content/themes/repo/newspack-theme/front-page.php:10
[08-Dec-2021 19:15:18 UTC] PHP   6. locate_template() /Users/dkoo/Local Sites/newspack/app/public/wp-includes/general-template.php:48
[08-Dec-2021 19:15:18 UTC] PHP   7. load_template() /Users/dkoo/Local Sites/newspack/app/public/wp-includes/template.php:716
[08-Dec-2021 19:15:18 UTC] PHP   8. require_once() /Users/dkoo/Local Sites/newspack/app/public/wp-includes/template.php:770
[08-Dec-2021 19:15:18 UTC] PHP   9. get_template_part() /Users/dkoo/Local Sites/newspack/app/public/wp-content/themes/repo/newspack-theme/header.php:39
[08-Dec-2021 19:15:18 UTC] PHP  10. locate_template() /Users/dkoo/Local Sites/newspack/app/public/wp-includes/general-template.php:204
[08-Dec-2021 19:15:18 UTC] PHP  11. load_template() /Users/dkoo/Local Sites/newspack/app/public/wp-includes/template.php:716
[08-Dec-2021 19:15:18 UTC] PHP  12. require() /Users/dkoo/Local Sites/newspack/app/public/wp-includes/template.php:772
[08-Dec-2021 19:15:18 UTC] PHP  13. dynamic_sidebar() /Users/dkoo/Local Sites/newspack/app/public/wp-content/themes/repo/newspack-theme/template-parts/header/mobile-sidebar.php:34
[08-Dec-2021 19:15:18 UTC] PHP  14. do_action() /Users/dkoo/Local Sites/newspack/app/public/wp-includes/widgets.php:724
[08-Dec-2021 19:15:18 UTC] PHP  15. WP_Hook->do_action() /Users/dkoo/Local Sites/newspack/app/public/wp-includes/plugin.php:470
[08-Dec-2021 19:15:18 UTC] PHP  16. WP_Hook->apply_filters() /Users/dkoo/Local Sites/newspack/app/public/wp-includes/class-wp-hook.php:327
[08-Dec-2021 19:15:18 UTC] PHP  17. Newspack_Ads_Sidebar_Placements::create_sidebar_before_action() /Users/dkoo/Local Sites/newspack/app/public/wp-includes/class-wp-hook.php:303
[08-Dec-2021 19:15:18 UTC] PHP  18. do_action() /Users/dkoo/Local Sites/newspack/app/public/wp-content/plugins/newspack-ads/includes/class-newspack-ads-sidebar-placements.php:46
[08-Dec-2021 19:15:18 UTC] PHP  19. WP_Hook->do_action() /Users/dkoo/Local Sites/newspack/app/public/wp-includes/plugin.php:470
[08-Dec-2021 19:15:18 UTC] PHP  20. WP_Hook->apply_filters() /Users/dkoo/Local Sites/newspack/app/public/wp-includes/class-wp-hook.php:327
[08-Dec-2021 19:15:18 UTC] PHP  21. Newspack_Ads_Placements::{closure:/Users/dkoo/Local Sites/newspack/app/public/wp-content/plugins/newspack-ads/includes/class-newspack-ads-placements.php:183-185}() /Users/dkoo/Local Sites/newspack/app/public/wp-includes/class-wp-hook.php:303
[08-Dec-2021 19:15:18 UTC] PHP  22. Newspack_Ads_Placements::inject_placement_ad() /Users/dkoo/Local Sites/newspack/app/public/wp-content/plugins/newspack-ads/includes/class-newspack-ads-placements.php:184
```

It's not happening at this moment on the fresh site, so I'm not sure what changed there, but I do see the error in the debug.log, so it was happening at some point during testing.

Reply:
It was missing another check to ensure that there is placement data for the hook. Fixed in 02b3606d4465841448a9a7e24e3bf225e0cdf122

Reply:
Confirmed, thanks!